jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -579,96 +585,34 @@ def _update_stretch_histogram(self, msg={}):
-            self.stretch_hist_nbins = 25
-
-        else:
-            hist_mark = self.stretch_histogram.marks[0]
-            hist_mark.sample = sub_data
-            # TODO: Let user change the number of bins?
-            # TODO: Let user set y-scale to log
+        hist_mark = self.stretch_histogram.marks['histogram']
+        hist_mark.sample = sub_data
+        hist_mark.bins = self.stretch_hist_nbins
+        hist_mark.send_state('sample')
 
         interval = PercentileInterval(95)
         if len(sub_data) > 0:
             hist_lims = interval.get_limits(sub_data)
             hist_mark.min, hist_mark.max = hist_lims
 
-        self._check_if_v_stretch_changed()
-
-    @observe('stretch_vmin_value', 'stretch_vmax_value')
-    def _check_if_v_stretch_changed(self, msg=None):
-        self._remove_histogram_marks()
-        self._add_histogram_marks()
+    @observe('stretch_vmin_value')
+    def _stretch_vmin_changed(self, msg=None):
+        self.stretch_histogram.marks['vmin'].x = [self.stretch_vmin_value] * 2
 
-    def _add_histogram_marks(self):
-        if self.stretch_histogram is None:
-            return
-        scales = {'x': self.stretch_histogram.axes[0].scale, 'y': bqplot.LinearScale()}
-        v_stretch_lines = []
-        # Set the viewer limits if they are None
-        if self.stretch_histogram.axes[0].scale.min is None:
-            self.stretch_histogram.axes[0].scale.min = self.stretch_histogram.marks[0].min
-        if self.stretch_histogram.axes[0].scale.max is None:
-            self.stretch_histogram.axes[0].scale.max = self.stretch_histogram.marks[0].max
-
-        # If the stretch limits are within the viewer limits, draw the HistogramMark
-        if self.stretch_vmin.value >= self.stretch_histogram.axes[0].scale.min:
-            vmin_lines = HistogramMark(min_max_value=[self.stretch_vmin.value, self.stretch_vmin.value], # noqa
-                                       scales=scales)
-            v_stretch_lines.append(vmin_lines)
-        if self.stretch_vmax.value <= self.stretch_histogram.axes[0].scale.max:
-            vmax_lines = HistogramMark(min_max_value=[self.stretch_vmax.value, self.stretch_vmax.value], # noqa
-                                       scales=scales)
-            v_stretch_lines.append(vmax_lines)
-        self.stretch_histogram.marks = self.stretch_histogram.marks + v_stretch_lines
-
-    def _remove_histogram_marks(self):
-        if self.stretch_histogram is None:
-            return
-        self.stretch_histogram.marks = [mark for mark in self.stretch_histogram.marks
-                                        if not isinstance(mark, HistogramMark)]
+    @observe('stretch_vmax_value')
+    def _stretch_vmax_changed(self, msg=None):
+        self.stretch_histogram.marks['vmax'].x = [self.stretch_vmax_value] * 2
 
-    @observe("stretch_hist_nbins")
-    def histogram_nbins_changed(self, msg):
+    @observe('stretch_hist_nbins')

```

Comment:
There is actually a movement to prefer double quotes but let's not open that can of worms here. But then again, no reason for this change?
```suggestion
    @observe("stretch_hist_nbins")
```

Reply:
I was just trying to be self-consistent between the three observes in a row - but no sense in adding to the diff (maybe I'll switch the new one instead).